connector/src/account_write_filter.rs

```diff
@@ -84,6 +84,31 @@ pub fn init(
                         },
                     );
                 }
+                Ok(FeedWrite::Snapshot(snapshot_write)) = account_write_queue_receiver.recv() => {
```

Comment:
is this kind of pattern matching on the future-result allowed? I'm worried that this runs two `account_write_queue_receiver.recv()` futures and then fails if whichever returns first doesn't match the pattern